src/teams.rs

```diff
@@ -4,6 +4,113 @@ use std::any::Any;
 use std::error::Error;
 use std::fmt;
 
+#[derive(Default, Serialize)]
+pub struct IndexData {
+    teams: Vec<IndexTeam>,
+    wgs: Vec<IndexTeam>,
+}
+
+#[derive(Serialize)]
+struct IndexTeam {
+    #[serde(flatten)]
+    team: Team,
+    url: String,
+}
+
+#[derive(Serialize)]
+pub struct PageData {
+    pub team: Team,
+    subteams: Vec<Team>,
+    wgs: Vec<Team>,
+}
+
+struct Data {
+    teams: Vec<Team>,
+}
+
+impl Data {
+    fn load() -> Result<Self, Box<Error>> {
+        Ok(Data {
+            teams: crate::cache::get::<Vec<Team>>(get_teams)?,
+        })
+    }
+
+    #[cfg(test)]
+    fn dummy(teams: Vec<Team>) -> Self {
+        Data { teams }
+    }
+
+    fn index_data(self) -> Result<IndexData, Box<Error>> {
+        let mut data = IndexData::default();
+
+        self.teams
+            .into_iter()
+            .filter(|team| team.website_data.is_some())
+            .filter(|team| team.subteam_of.is_none())
+            .map(|team| IndexTeam {
+                url: format!(
+                    "{}/{}",
+                    kind_to_str(team.kind),
+                    team.website_data.as_ref().unwrap().page
+                ),
+                team,
+            })
+            .for_each(|team| match team.team.kind {
+                TeamKind::Team => data.teams.push(team),
+                TeamKind::WorkingGroup => data.wgs.push(team),
+            });
+
+        data.teams
+            .sort_by_key(|team| -team.team.website_data.as_ref().unwrap().weight);
```

Comment:
I personally would like this to be replaced with e.g. the below or otherwise made to not read as `team.team` which just looks quite odd with relatively little gain I think.

```suggestion
            .sort_by_key(|idx_team| -idx_team.team.website_data.as_ref().unwrap().weight);
```